server/filestore.go

```diff
@@ -5348,11 +5348,14 @@ func decodeConsumerState(buf []byte) (*ConsumerState, error) {
 				dseq = readSeq()
 			}
 			ts := readTimeStamp()
-			if sseq == 0 || ts == -1 {
+			if ts == -1 {
 				return nil, errCorruptState
 			}
 			// Adjust seq back.
 			sseq += state.AckFloor.Stream
+			if sseq == 0 {
```

Comment:
@derekcollison I don't know if `dseq == 0` is valid or not. If not valid, then this test could be moved down few lines with `if sseq == 0 || dseq == 0 { return nil, errCorruptState}`

Reply:
Stream and Consumer can only be 0 with no activity at all, they are 1s based meaning the first message for a stream and a consumer is always 1.

Reply:
Sorry, not sure I understand. Should we check for both sseq and dseq == 0 meaning corrupt state or not? You wrote "stream and consumer can only be 0 with no activity at all", but that does not tell me if it is possible to have a consumer state with those values at 0. Seem to me that it is not since I believe that stream sequence (and therefore consumer?) starts at 1, and there can't be a *pending* with sseq/dseq == 0, right?

Reply:
Just sseq